### PR TITLE
Improve docs

### PR DESCRIPTION
**Pull request by lvwerra**
The goal of this PR is to improve the overall docs:
- add links to blogs (maybe we could add thumbnails?)
- improve readme
- how to generate and which metrics to look for in PPO
- how to use a trained model for inference (#599)

### CODE REVIEW

File: docs/source/how_to_train.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Training FAQ
+
+## What Metrics Should I Look at?
+
+When performing classical supervised fine-tuning of language models, the loss (especially the validation loss) serves as a good indicator of the training progress. However, in Reinforcement Learning (RL), the loss becomes less informative about the model's performance, and its value may fluctuate while the actual performance improves.
+
+To address this, we recommend focusing on two key metrics:
+
+**Mean Reward**: The primary goal is to maximize the reward achieved by the model during RL training.
+**Objective KL Divergence**: KL divergence (Kullback-Leibler divergence) measures the dissimilarity between two probability distributions. In the context of RL training, we use it to quantify the difference between the current model and a reference model. Ideally, we want to keep the KL divergence between 0 and 10 to ensure the model's generated text remains close to what the reference model produces.
```
Comment (vwxyzjn) [inline docs/source/how_to_train.md:10]: Maybe we can merge this and https://github.com/lvwerra/trl/blob/main/docs/source/logging.mdx? I think another objective is entropy, where we would want the model to be as chaotic as possible.

File: docs/source/how_to_train.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+## Why Do We Use a Reference Model, and What's the Purpose of KL Divergence?
+
+When training RL models, optimizing solely for reward may lead to unexpected behaviors, where the model exploits the environment in ways that don't align with good language generation. In the case of RLHF, we use a reward model trained to predict whether a generated text is highly ranked by humans.
+
+However, the RL model being optimized against the reward model may learn patterns that yield high reward but do not represent good language. This can result in extreme cases where the model generates texts with excessive exclamation marks or emojis to maximize the reward. In some worst-case scenarios, the model may generate patterns completely unrelated to natural language yet receive high rewards, similar to adversarial attacks.
```
Comment (vwxyzjn) [inline docs/source/how_to_train.md:16]: We can probabaly give some examples/references. E.g., Table 10 from https://arxiv.org/pdf/1909.08593.pdf
image

File: docs/source/how_to_train.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+- **Start simple**: Try to start with a minimal example and build complexity from there. Your use-case might require for example a complicated reward function consisting of many different rewards - try to use one signal first and see if you can optimize that and then add more complexity after that.
+- **Inspect the generations**: It's always a good idea to inspect what the model is generating. Maybe there is a big in your post-processing or your prompt. Due to bad settings you might cut-off generations too soon. These things are very hard to see on the metrics but very obvious if you look at the generations.
+- **Inspect the reward model**: If you reward is not improving over time maybe there's an issue with the reward model. You can look at extreme cases to see if it does what it should: e.g. in the sentiment case you can check if simple positive and negative examples really get different rewards. And you can look at the distribution of your dataset. Finally, maybe the reward is dominated by the query which the model can't affect so you might need to normalize this (e.g. reward of query+response minus reward of the query).
```
Comment (vwxyzjn) [inline docs/source/how_to_train.md:57]: This is very nicely done!

File: docs/source/use_model.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+Once you have trained a model using either the SFTTrainer, PPOTrainer, or DPOTrainer, you will have a fine-tuned model that can be used for text generation. In this section, we'll walk through the process of loading the fine-tuned model and generating text. If you need to run an inference server with the trained model, you can explore libraries such as [`text-generation-inference`](https://github.com/huggingface/text-generation-inference).
+
+## Load and Generate
+
+If you have fine-tuned a model fully, meaning without the use of PEFT you can simply load it like any other language model in transformers. E.g. the value head that was trained during the PPO training is no longer needed and if you load the model with the original transformer class it will be ignored:
+
+```python
+from transformers import AutoTokenizer, AutoModelForCausalLM
+
+model_name_or_path = "path/to/your/model/or/name/on/hub"
```
Comment (vwxyzjn) [inline docs/source/use_model.md:12]: Maybe we can provide a working example and put  `# "path/to/your/model/or/name/on/hub"`

**Review by vwxyzjn — APPROVED** [review]
Looks really good! Thanks @lvwerra. I left some minor comments.

File: docs/source/how_to_train.md (hunk 3 above)
Comment (lvwerra) [reply docs/source/how_to_train.md:57]: Thanks :)

File: docs/source/how_to_train.md (hunk 2 above)
Comment (lvwerra) [reply docs/source/how_to_train.md:18]: Good idea, added!

File: docs/source/how_to_train.md (hunk 1 above)
Comment (lvwerra) [reply docs/source/how_to_train.md]: I added a link to the logging page, what do you think?

**Pull request merged**